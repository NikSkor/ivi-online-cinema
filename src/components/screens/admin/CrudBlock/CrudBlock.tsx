import { FC, useState } from "react";
import style from './CrudBlock.module.scss';
import Link from "next/link";
import { useRouter } from "next/router";

interface ICrud {
  item: {
    id: number,
    name: string,
    enName: string|null
  },
  adress: string,
  children: React.ReactNode
}

const CrudBlock: FC<ICrud> = ({item, adress, children}) => {

const locale = useRouter().locale;

  return (
      <li key={item.id} className={style.item}>
      {locale !== 'ru' && item.enName !== null
        ? 
        <h4 className={style.itemTitle}>{item.enName}</h4>
        :
        <h4 className={style.itemTitle}>{item.name}</h4>
      }
      
      <div className={style.actionBlock}>
        {locale === 'ru'
          ? 
          <Link href={`${adress}${item.id}`} className={style.actionBtn} >
            <p data-id={item.id}>Редактировать</p>
          </Link>
          :
          <Link href={`${adress}${item.id}`} className={style.actionBtn} >
            <p data-id={item.id}>Redact</p>
          </Link>
        }
        
      <button className={style.actionBtn} onClick={()=>{}}>
        {children}
      </button>
      </div>
    </li>
  )

}

export default CrudBlock;