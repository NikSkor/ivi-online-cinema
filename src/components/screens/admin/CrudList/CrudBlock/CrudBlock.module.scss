@import '../../../../../styles/vars.scss';

.item {
  display: flex;
  align-items: center;
  // gap: 50px;
  justify-content: space-between;
}

.itemTitle {
  font-weight: 500;
  font-size: 16px;
  // width: 800px;

  @include mobile {
    font-size: 12px;
    max-width: 160px;
  }
}

.actionBlock {
  display: flex;
  gap: 30px;

  @include laptop {
    gap: 10px;
  }

  @include mobile {
    gap: 5px;
  }
}

.actionBtn {
  font-weight: 500;
  font-size: 12px;
  justify-self: flex-end;
  min-width: 101px;

  position: relative;
  display: inline-flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  height: 40px;
  padding: 12px 12px;

  border: solid 1px $btn_bg;
  background-color: $btn_bg;
  border-radius: 8px;

  cursor: pointer;
  text-decoration: none;
  color: $primary;

  transition-property: border-color, background-color;
  transition-duration: .4s;

  &:not(:last-child) {
    margin-right: 12px;
  }

  &.square {
    width: 40px;
    padding: 0;
  }

  &:hover {
    background-color: $btn_hover;
    border-color: $btn_hover;
  }

  @include mobile {
    font-size: 10px;
  }
}

.minBtn {
  min-width: 40px;
}

.visibleBtn {
  display: none;

  @include mobile {
    display: block;
  }
}

.invisibleBtn {
  display: block;

  @include mobile {
    display: none;
  }
}