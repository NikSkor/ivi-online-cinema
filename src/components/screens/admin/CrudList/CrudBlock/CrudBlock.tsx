import { FC, useState } from "react";
import style from './CrudBlock.module.scss';
import Link from "next/link";
import { useRouter } from "next/router";
import Image from "next/image";
import trashImg from '../../../../../../public/trash.svg';
import editImg from '../../../../../../public/editIcon.svg';


interface ICrud {
  item: {
    id: number,
    name: string,
    enName: string|null
  },
  adress: string,
  children: React.ReactNode
}

const CrudBlock: FC<ICrud> = ({item, adress, children}) => {

const locale = useRouter().locale;

  return (
      <li key={item.id} className={style.item}>
      {locale !== 'ru' && item.enName !== null
        ? 
        <h4 className={style.itemTitle}>{item.enName}</h4>
        :
        <h4 className={style.itemTitle}>{item.name}</h4>
      }
      
      <div className={style.actionBlock}>
        {locale === 'ru'
          ? 
          <Link href={`${adress}${item.id}`} >
            <button className={`${style.actionBtn} ${style.invisibleBtn}`}>Редактировать</button>
          </Link>
          :
          <Link href={`${adress}${item.id}`}>
            <button className={`${style.actionBtn} ${style.invisibleBtn}`}>Redact</button>
          </Link>
        }
      <Link href={`${adress}${item.id}`}>
          <button className={`${style.actionBtn} ${style.minBtn} ${style.visibleBtn}`}>
            <Image src={editImg} width={20} height={20} alt="Значок редактирования"/>
          </button>
        </Link>
      <button className={`${style.actionBtn} ${style.minBtn}`} onClick={()=>{}}>
        <Image src={trashImg} width={19} height={19} alt="Значок очистки"/>
      </button>
      </div>
    </li>
  )

}

export default CrudBlock;