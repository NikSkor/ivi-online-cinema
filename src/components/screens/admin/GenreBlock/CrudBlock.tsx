import { FC, useState } from "react";
import style from './CrudBlock.module.scss';
import Link from "next/link";

interface ICrud {
  item: {
    id: number,
    name: string
  },
  adress: string,
  children: React.ReactNode
}

const CrudBlock: FC<ICrud> = ({item, adress, children}) => {

  return (
      <li key={item.id} className={style.item}>
      <h4 className={style.itemTitle}>{item.name}</h4>
      <div className={style.actionBlock}>
        <Link href={`${adress}${item.id}`} className={style.actionBtn} >
        <p data-id={item.id}>Редактировать</p>
      </Link>
      <button className={style.actionBtn} onClick={()=>{}}>
        {children}
      </button>
      </div>
    </li>
  )

}

export default CrudBlock;