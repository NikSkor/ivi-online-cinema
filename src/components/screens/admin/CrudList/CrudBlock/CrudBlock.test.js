import CrudBlock from '@/components/screens/admin/CrudList/CrudBlock/CrudBlock';
import { cleanup, render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import mockRouter from 'next-router-mock';
import '@testing-library/jest-dom/extend-expect';

jest.mock('next/router', () => require('next-router-mock'));

describe('test CrudBlock component', () => {
  afterEach(cleanup);

  test('there are the buttons', () => {
    render(
      <CrudBlock
        key={1}
        item={{
          id: 1,
          name: 'Имя',
          enName: 'Name',
        }}
        adress={'/foo'}
      />
    );
    const btns = screen.getAllByRole('button');
    btns.forEach((item) => {
      expect(item).toBeInTheDocument();
      expect(item).toMatchSnapshot();
    });
  });

  test('btn go to edit page', () => {
    mockRouter.push('/foo');

    render(
      <CrudBlock
        key={1}
        item={{
          id: 1,
          name: 'Имя',
          enName: 'Name',
        }}
        adress={'/'}
      />
    );

    fireEvent.click(screen.getByText('Redact'));

    expect(mockRouter).toMatchObject({
      asPath: '/foo',
      pathname: '/foo',
    });
  });

  test('component renders props correctly, truthy values', () => {
    const tree = renderer
      .create(
        <CrudBlock
          key={1}
          item={{
            id: 1,
            name: 'Имя',
            enName: 'Name',
          }}
          adress={'/foo'}
        ></CrudBlock>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('component renders props correctly, falsy values', () => {
    const tree = renderer
      .create(
        <CrudBlock
          key={''}
          item={{
            id: '',
            name: '',
            enName: '',
          }}
          adress={''}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
