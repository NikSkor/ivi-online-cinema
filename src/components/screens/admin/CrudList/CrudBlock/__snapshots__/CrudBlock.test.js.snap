// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CrudBlock component component renders props correctly, falsy values 1`] = `
<li
  className="item"
>
  <h4
    className="itemTitle"
  />
  <div
    className="actionBlock"
  >
    <a
      href=""
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <button
        className="actionBtn invisibleBtn"
      >
        Redact
      </button>
    </a>
    <a
      href=""
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <button
        className="actionBtn minBtn visibleBtn"
      >
        <img
          alt="Значок редактирования"
          data-nimg="1"
          decoding="async"
          height={20}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
          srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={20}
        />
      </button>
    </a>
    <button
      className="actionBtn minBtn"
      onClick={[Function]}
    >
      <img
        alt="Значок очистки"
        data-nimg="1"
        decoding="async"
        height={19}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={19}
      />
    </button>
  </div>
</li>
`;

exports[`test CrudBlock component component renders props correctly, truthy values 1`] = `
<li
  className="item"
>
  <h4
    className="itemTitle"
  >
    Name
  </h4>
  <div
    className="actionBlock"
  >
    <a
      href="/foo1"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <button
        className="actionBtn invisibleBtn"
      >
        Redact
      </button>
    </a>
    <a
      href="/foo1"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <button
        className="actionBtn minBtn visibleBtn"
      >
        <img
          alt="Значок редактирования"
          data-nimg="1"
          decoding="async"
          height={20}
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
          srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={20}
        />
      </button>
    </a>
    <button
      className="actionBtn minBtn"
      onClick={[Function]}
    >
      <img
        alt="Значок очистки"
        data-nimg="1"
        decoding="async"
        height={19}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
        style={
          {
            "color": "transparent",
          }
        }
        width={19}
      />
    </button>
  </div>
</li>
`;

exports[`test CrudBlock component there are the buttons 1`] = `
<button
  class="actionBtn invisibleBtn"
>
  Redact
</button>
`;

exports[`test CrudBlock component there are the buttons 2`] = `
<button
  class="actionBtn minBtn visibleBtn"
>
  <img
    alt="Значок редактирования"
    data-nimg="1"
    decoding="async"
    height="20"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
    style="color: transparent;"
    width="20"
  />
</button>
`;

exports[`test CrudBlock component there are the buttons 3`] = `
<button
  class="actionBtn minBtn"
>
  <img
    alt="Значок очистки"
    data-nimg="1"
    decoding="async"
    height="19"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
    style="color: transparent;"
    width="19"
  />
</button>
`;
